compiler/src/dmd/common/blake3.d

```diff
@@ -0,0 +1,346 @@
+module dmd.common.blake3;
+// based on https://github.com/oconnor663/blake3_reference_impl_c/blob/main/reference_impl.c
+
+nothrow:
+
+/**
+ * Implementation of Blake 3 hash function with streaming disabled
+ * meaning we hash the whole buffer at once.
+ * Input is split into 1KB Chunks which could be hashed independently.
+ * That said, in the compiler I expect almost all inputs will be 1 chunk.
+ *
+ * Chunks get split into 64B Blocks which get hashed and then mixed together
+ *
+ * Params:
+ *     data = byte array to hash
+ * Returns: Blake 3 hash of data
+ **/
+public ubyte[32] blake3(const ubyte[] data)
+{
+
+
+    ChunkState state;
+    CVStack cvStack;
+
+    size_t cursor = 0;
```

Comment:
Also, I would group these 3 vars. There is no need for the empty line between them.